autoPyTorch/pipeline/base_pipeline.py

```diff
@@ -425,52 +425,66 @@ def _check_search_space_updates(self, include: Optional[Dict[str, Any]],
                             if choice in exclude[update.node_name]:
                                 raise ValueError("Found {} in exclude".format(choice))
                         if choice not in components.keys():
-                            raise ValueError("Unknown hyperparameter for choice {}. "
-                                             "Expected update hyperparameter "
-                                             "to be in {} got {}".format(node.__class__.__name__,
-                                                                         components.keys(), choice))
+                            warnings.warn("Unknown hyperparameter for choice {}. "
+                                          "Expected update hyperparameter "
+                                          "to be in {} got {}. "
+                                          "This update {} will be removed".format(node.__class__.__name__,
+                                                                                  components.keys(), choice, update),
+                                          UserWarning)
+                            self.search_space_updates.updates.remove(update)
```

Comment:
Can you call the remove outside of the foreach loop? Like I don't think it is a good idea to update the list while iterating on it

Reply:
okay, I'll do so